extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -840,18 +847,24 @@ export class AmpA4A extends AMP.BaseElement {
    */
   renderViaSafeFrame_(creativeBody) {
     this.emitLifecycleEvent('renderSafeFrameStart');
-    return utf8Decode(creativeBody).then(creative => {
-      /** @const {!Element} */
-      const iframe = createElementWithAttributes(
-          /** @type {!Document} */(this.element.ownerDocument),
-          'iframe', Object.assign({
-            'height': this.element.getAttribute('height'),
-            'width': this.element.getAttribute('width'),
-            'src': SAFEFRAME_IMPL_PATH + '?n=0',
-            'name': `${SAFEFRAME_VERSION};${creative.length};${creative}`,
-          }, SHARED_IFRAME_PROPERTIES));
-      return this.iframeRenderHelper_(iframe);
+    let decodeErr;
+    const creative = tryUtf8Decode(creativeBody, err => {
```

Comment:
well, it's certainly cleaner when inside a promise, but uglier with try-catch when used outside of a promise. I am not too opinionated though, so I will change it since you hate it, but should it always throw? Should we do the optional `fatal: true` like the native decoder? 

The part of try-semantics I like is that it makes it very clear what the behaviour is as it clearly tells the user it won't throw.